dandi/tests/test_validate.py

```diff
@@ -1,34 +1,63 @@
-from glob import glob
 import json
 import os
 import pathlib
 
 import pytest
 
 from .fixtures import BIDS_TESTDATA_SELECTION
-from ..validate import validate_bids
+from ..validate import validate
+
+
+def test_validate_nwb_error(simple3_nwb):
+    """
+    Do we fail on critical NWB validation errors?
+    """
+
+    validation_result = validate(simple3_nwb)
+    assert len([i for i in validation_result if i.severity]) > 0
 
 
 @pytest.mark.parametrize("dataset", BIDS_TESTDATA_SELECTION)
 def test_validate_bids(bids_examples, tmp_path, dataset):
 
     selected_dataset = os.path.join(bids_examples, dataset)
-    validation_result = validate_bids(selected_dataset, report=True)
+    validation_result = validate(selected_dataset, report=True)
     for i in validation_result:
         assert i.severity is None
 
 
-def test_report_path(bids_examples, tmp_path):
+def test_validate_bids_onefile(bids_error_examples, tmp_path):
+    """
+    Dedicated test using to single-file validation.
 
-    report_path = os.path.join(tmp_path, "inplace_bids-validator-report.log")
-    selected_dataset = os.path.join(bids_examples, BIDS_TESTDATA_SELECTION[0])
-    _ = validate_bids(
-        selected_dataset,
-        report_path=report_path,
+    Notes
+    -----
+    * Due to the dataset-wide scope of BIDS, issues have arisen and can potentially arise again
+    with single-file handling. Best to keep this in to always make sure.
+    * This can be further automated thanks to the upstream `.ERRORS.json` convention to be
+    performed on all error datasets, but that might be overkill since we test the datasets as
+    a whole anyway.
+    """
+
+    selected_dataset, error_file = (
+        "invalid_asl003",
+        "sub-Sub1/perf/sub-Sub1_headshape.jpg",
     )
 
-    # Check if a report is being produced.
-    assert len(glob(report_path)) == 1
+    bids_file_path = os.path.join(bids_error_examples, selected_dataset, error_file)
+    bids_dataset_path = pathlib.Path(
+        os.path.join(bids_error_examples, selected_dataset)
```

Comment:
```suggestion
        bids_error_examples, selected_dataset
```

No need to `join` inside the `Path` constructor.